### PR TITLE
Follow up on recent Minuit 2 code changes

### PR DESCRIPTION
**Pull request by guitargeek**
See [this comment](https://github.com/root-project/root/pull/16491#issuecomment-2364275819) for more context.

This PR will work with ROOT <=6.32 or with ROOT master once this PR is merged:
https://github.com/root-project/root/pull/16491

@smuzaffar, FYI, sorry again for the trouble!

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
hold

we need to wait for https://github.com/root-project/root/pull/16491 to be tested for ROOT6 IBs first

File: OnlineDB/CSCCondDB/src/CSCFitAFEBThr.cc [hunk 1]
```diff
@@ -130,12 +129,8 @@ bool CSCFitAFEBThr::ThresholdNoise(const std::vector<float>& inputx,
   //                                               <<" "<<ery[i]<<std::endl;
 
   /// Fit  as 1D, <=500 iterations, edm=10**-5 (->0.1)
-#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 33, 1)
-  FunctionMinimum fmin =
-      theFitter->Minimize(*theOBJfun, std::span<double>(parinit), std::span<double>(erparinit), 1, 500, 0.1);
-#else
-  FunctionMinimum fmin = theFitter->Minimize(*theOBJfun, parinit, erparinit, 1, 500, 0.1);
-#endif
+  FunctionMinimum fmin = theFitter->Minimize(*theOBJfun, {parinit, erparinit}, MnStrategy{1}, 500, 0.1);
```
[the remark this answers is not in the document]
Comment (iarspider) [reply OnlineDB/CSCCondDB/src/CSCFitAFEBThr.cc:132]: @smuzaffar @guitargeek adding `#include <Minuit2/MnStrategy.h>` may help.

[the remark this answers is not in the document]
Comment (guitargeek) [reply OnlineDB/CSCCondDB/src/CSCFitAFEBThr.cc:132]: Good catch! I'll add this.

[the remark this answers is not in the document]
Comment (guitargeek) [reply OnlineDB/CSCCondDB/src/CSCFitAFEBThr.cc:133]: It was already added by another PR now (thanks!):
https://github.com/cms-sw/cmssw/pull/46195

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test with #46195

**Comment by guitargeek** [comment]
Still ambiguous? Ok so the explicit `std::span` are needed then. I'll update the PR

**Comment by guitargeek** [comment]
It should work now! Sorry for doing these PRs blindly. I have no reasonably fast machine with CVMFS access to do CMSSW development anymore.

**Comment by smuzaffar** [comment]
please test

**Comment by guitargeek** [comment]
Ok I don't get it. I'll try to compile myself on lxplus9 first.

**Comment by guitargeek** [comment]
Alright, there was some problem with the overload resolution in case of ROOT 6.30. I now construct the `MnUserParameterState` explicitly, then it should work with all ROOT versions. I tested it with the most recent IB: `CMSSW_14_2_X_2024-10-06-0000`.

**Comment by smuzaffar** [comment]
Please test

**Comment by jfernan2** [comment]
]1

**Comment by smuzaffar** [comment]
@cms-sw/alca-l2 @cms-sw/db-l2 can you please review this? This is revert of #45965 PR . This is needed to integrate latest root 6.33/master changes in to our ROOT6_X IBs

**Comment by perrotta** [comment]
+1

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**